File: Haskell/ProjectEuler002.hs
fib :: Int -> Int
fib 0 = 1
fib 1 = 2
fib n = fib (n-1) + fib (n-2)

fiblist :: Int -> Int -> [Int]
fiblist n limit
	| fib n <= limit = (fib n) : fiblist (n+1) limit
	| otherwise = []

solve :: Int -> Int
solve n = sum $ filter (\x -> even x) (fiblist 0 n)

File: C++/ProjectEuler002.cpp
#include <iostream>

constexpr auto solve(const int limit)
{
	auto first = 1;
	auto next = 2;
	auto sum = 0;
	while (first <= limit) 
	{
		if (first % 2 == 0) 
			sum += first;
		auto temp = first + next;
		first = next;
		next = temp;
	}
	return sum;
}

auto main() -> int
{
	std::cout << solve(4000000) << std::endl;
	return 0;
}
